pkg/rbac/parser.go

```diff
@@ -44,8 +48,62 @@ type Rule struct {
 	URLs      []string `marker:"urls,optional"`
 }
 
+// target generates a string representing the resources and/or non-resources a Rule applies.
+func (r *Rule) target() string {
+	r.cleanUp()
+	s := []string{
```

Comment:
I don't think this is sufficient -- I think we should spearately groups from resources from urls with a different character than we used to separate the individual items.

Reply:
Removed this method and use ruleKey as the map key.